command/agent/acl_endpoint.go

```diff
@@ -138,8 +138,14 @@ func (s *HTTPServer) ACLTokenBootstrap(resp http.ResponseWriter, req *http.Reque
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 
-	// Format the request
-	args := structs.ACLTokenBootstrapRequest{}
+	var args structs.ACLTokenBootstrapRequest
+
+	if req.Body != nil {
```

Comment:
The reason i went for body is that the content length is not reliable and sometimes returns 0 even when the body contains the full token.
So I went for what I thought would be a reliable option.

Reply:
@schmichael do you want me to change anything here?

Reply:
Yes, `Reqest.Body` [cannot be nil in this server context](https://pkg.go.dev/net/http#Request), so this check is never hit.

`nomad acl bootstrap` now _always_ sends a body, which is fine, but means if you're testing with the CLI it will always work. Your `command/agent/acl_endpoint_test.go` test doesn't test the empty body case.

If you add a test case for an empty body you will see that prior to your change an empty body to this HTTP API works, but after your change the HTTP API returns a 400 on an empty body.

You can also test this with curl:

```
# Should work but fails:
~$ curl -v -XPOST localhost:4646/v1/acl/bootstrap
> POST /v1/acl/bootstrap HTTP/1.1
> Host: localhost:4646
> User-Agent: curl/7.68.0
> Accept: */*
>
< HTTP/1.1 400 Bad Request
< Vary: Accept-Encoding
< Date: Tue, 31 May 2022 21:45:53 GMT
< Content-Length: 21
< Content-Type: text/plain; charset=utf-8
<
Request body is empty
```